mla/models.py

```diff
@@ -290,209 +291,217 @@ def background_dec_spline(self) -> scipy.interpolate.UnivariateSpline:
+        bg_h /= np.sum(bg_h, axis=1)[:, None]
         return bg_h
-    
-    def _create_sob_ratio(self, *args, **kwargs)-> None:
+
+    def _create_sob_ratio(self, *args, **kwargs) -> None:
         """Create the SOB map with a spectrum
         """
         bins = np.array([self._sin_dec_bins, self._log_energy_bins])
-        bin_centers = bins[1,:-1] + np.diff(bins[1])/2
-        sig_w = self.reduced_sim['ow'] * self.spectrum(self.reduced_sim['trueE'])
-        sig_h, xedges, yedges = np.histogram2d(
-            self.reduced_sim['sindec'],
-            self.reduced_sim['logE'],
-            bins=bins,
-            weights=sig_w,
-            density=True)
-        
+        bin_centers = bins[1, :-1] + np.diff(bins[1]) / 2
+        sig_w = self.reduced_sim['ow'] * self.spectrum(
+            self.reduced_sim['trueE'])
+        sig_h, _, _ = np.histogram2d(self.reduced_sim['sindec'],
+                                     self.reduced_sim['logE'], bins=bins,
+                                     weights=sig_w, density=True)
+
         # Normalize histograms by dec band
-        
-        sig_h /= np.sum(sig_h,axis=1)[:,None]
-        if 'k' not in kwargs: kwargs['k'] = 1
-        if 's' not in kwargs: kwargs['s'] = 0
-        if 'ext' not in kwargs: kwargs['ext'] = 3       
+
+        sig_h /= np.sum(sig_h, axis=1)[:, None]
+        if 'k' not in kwargs:
+            kwargs['k'] = 1
+        if 's' not in kwargs:
+            kwargs['s'] = 0
+        if 'ext' not in kwargs:
+            kwargs['ext'] = 3
         ratio = sig_h / self._background_sob_map
         for i in range(ratio.shape[0]):
             # Pick out the values we want to use.
             # We explicitly want to avoid NaNs and infinities
             values = ratio[i]
-            good = np.isfinite(values) & (values>0)
-            x, y = bin_centers[good], values[good]
+            good = np.isfinite(values) & (values > 0)
+            x_good, y_good = bin_centers[good], values[good]
 
             # Do a linear interpolation across the energy range
-            if len(x) > 1:
-                spline = scipy.interpolate.UnivariateSpline(x, y, *args, **kwargs)
-                ratio[i]=spline(bin_centers)
-            elif len(x)==1:
-                ratio[i]=y
+            if len(x_good) > 1:
+                spline = scipy.interpolate.UnivariateSpline(x_good, y_good,
+                                                            *args, **kwargs)
+                ratio[i] = spline(bin_centers)
+            elif len(x_good) == 1:
+                ratio[i] = y_good
             else:
-                ratio[i]=0
+                ratio[i] = 0
         return ratio
-    
-    
-    def get_energy_sob(self, events: np.ndarray) -> List[scipy.interpolate.UnivariateSpline]:
-        """Gets the signal-over-background vs. gamma required for each event and specific .
-        
+
+    def get_energy_sob(self, events: np.ndarray
+    ) -> List[scipy.interpolate.UnivariateSpline]:
```

Comment:
Sorry I made a mistake here. It should return np.ndarray instead of List[scipy.interpolate.UnivariateSpline] .